src/export/CodeSystemExporter.ts

```diff
@@ -26,19 +27,35 @@ export class CodeSystemExporter {
 
   private setConcepts(codeSystem: CodeSystem, concepts: ConceptRule[]): void {
     if (concepts.length > 0) {
-      codeSystem.concept = concepts.map(concept => {
-        const codeSystemConcept: CodeSystemConcept = { code: concept.code };
-        if (concept.display) codeSystemConcept.display = concept.display;
-        if (concept.definition) codeSystemConcept.definition = concept.definition;
-        return codeSystemConcept;
+      codeSystem.concept = [];
+      concepts.forEach(concept => {
+        let conceptContainer = codeSystem.concept;
+        const newConcept: CodeSystemConcept = { code: concept.code };
+        if (concept.display) {
+          newConcept.display = concept.display;
+        }
+        if (concept.definition) {
+          newConcept.definition = concept.definition;
+        }
+        concept.hierarchy.forEach(ancestorCode => {
+          const ancestorConcept = conceptContainer.find(
+            ancestorConcept => ancestorConcept.code === ancestorCode
+          );
+          if (!ancestorConcept.concept) {
+            ancestorConcept.concept = [];
+          }
```

Comment:
I tested manually and it looks like import process prevents us from getting this far with an invalid hierarchy.  And, in fact, it's the FSH CodeSystem that prevents it.  Still... I wonder, should we be defensive here just in case?  In theory, if someone constructed a `FshCodeSystem` by pushing to the `rules` array directly (instead of through `addConcept`), we could get to that case.